### PR TITLE
feat(ui): apply diamond architecture to the remained or missed components

### LINKED ISSUE #807: UI: apply diamond architecture to the remained or missed components
The diamond architecture has been applied to each pages of ui, but still some components(including monitor page) are untouched. and the ui codebase needs to be checked at the point of whole implementation of new architecture

**list of components**
- [inventory.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/layout/inventory.tsx)
- [metapool-monitor.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/monitor/metapool-monitor.tsx)
- [price-monitor.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/monitor/price-monitor.tsx)
- [token-monitor.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/monitor/token-monitor.tsx)
- [debt-coupon-deposit.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/redeem/debt-coupon-deposit.tsx)
- [ucr-nft-redeem.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/redeem/ucr-nft-redeem.tsx)
- [ucr-redeem.tsx](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/dapp/components/redeem/ucr-redeem.tsx)

Also the old names used in ui still not well compatible with new architecture.

We have a convention of using full names for tokens (ex: `uad => Dollar, ucr => Credit, ubq => Governance`). This issue implies refactoring naming as well.

### PR DESCRIPTION
**Pull request by bojan07**
Resolves #807

### CODE REVIEW

**Comment by bojan07** [comment]
@rndquu @molecula451 PR is ready for a review. I have debugged updates, and works well in my side.

**Comment by molecula451** [comment]
Can You make sure there are no linter warnings those when You build the dapp `yarn build:dapp` and pop

**Comment by bojan07** [comment]
> Can You make sure there are no linter warnings those when You build the dapp `yarn build:dapp` and pop

image
Build was successful. A few prettier warnings

**Comment by molecula451** [comment]
> > Can You make sure there are no linter warnings those when You build the dapp `yarn build:dapp` and pop
> 
> image Build was successful. A few prettier warnings

those are known, you may ignore em, is this PR really ready? you seem to be asking things at #807

**Comment by bojan07** [comment]
> > > Can You make sure there are no linter warnings those when You build the dapp `yarn build:dapp` and pop
> > 
> > 
> > image Build was successful. A few prettier warnings
> 
> those are known, you may ignore em, is this PR really ready? you seem to be asking things at #807

Yeah, except for `uAR` replacement, all is ready
And it was only used in [this file](https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/debt-coupon.tsx)

**Comment by molecula451** [comment]
are you testing the changes with the whole connection `yarn start:anvil && yarn deploy:development`

**Comment by bojan07** [comment]
> are you testing the changes with the whole connection `yarn start:anvil && yarn deploy:development`

Yeah, I am.

**Review by rndquu — CHANGES_REQUESTED** [review]
Screenshot 2023-10-19 at 10 16 31

The [inventory component](https://github.com/ubiquity/ubiquity-dollar/blob/73e19b308f4f622d6ce28ca0aedddfdd3c37a110/packages/dapp/components/layout/inventory.tsx) is rendered with errors if balances are set. Pls make sure that all of the components are rendered without errors.

**Comment by bojan07** [comment]
> The [inventory component](https://github.com/ubiquity/ubiquity-dollar/blob/73e19b308f4f622d6ce28ca0aedddfdd3c37a110/packages/dapp/components/layout/inventory.tsx) is rendered with errors if balances are set. Pls make sure that all of the components are rendered without errors.

I have already test it by setting the balances manually when debugging. How to set balances and check? I'm testing with first wallet on anvil testnet imported in Metamask

**Comment by rndquu** [comment]
> > The [inventory component](https://github.com/ubiquity/ubiquity-dollar/blob/73e19b308f4f622d6ce28ca0aedddfdd3c37a110/packages/dapp/components/layout/inventory.tsx) is rendered with errors if balances are set. Pls make sure that all of the components are rendered without errors.
> 
> I have already test it by setting the balances manually when debugging. How to set balances and check? I'm testing with first wallet on anvil testnet imported in Metamask

Simply set the balances manually in the code

**Comment by bojan07** [comment]
> > > The [inventory component](https://github.com/ubiquity/ubiquity-dollar/blob/73e19b308f4f622d6ce28ca0aedddfdd3c37a110/packages/dapp/components/layout/inventory.tsx) is rendered with errors if balances are set. Pls make sure that all of the components are rendered without errors.
> > 
> > 
> > I have already test it by setting the balances manually when debugging. How to set balances and check? I'm testing with first wallet on anvil testnet imported in Metamask
> 
> Simply set the balances manually in the code

Yeah, how to set BigNumber manually?

**Comment by bojan07** [comment]
> > > > The [inventory component](https://github.com/ubiquity/ubiquity-dollar/blob/73e19b308f4f622d6ce28ca0aedddfdd3c37a110/packages/dapp/components/layout/inventory.tsx) is rendered with errors if balances are set. Pls make sure that all of the components are rendered without errors.
> > > 
> > > 
> > > I have already test it by setting the balances manually when debugging. How to set balances and check? I'm testing with first wallet on anvil testnet imported in Metamask
> > 
> > 
> > Simply set the balances manually in the code
> 
> Yeah, how to set BigNumber manually?

resolved, `BigNumber.from()` works

**Comment by bojan07** [comment]
image
Fixed inventory error. Nice review. Now ready for a review again.

**Review by rndquu — CHANGES_REQUESTED** [review]
There is an error in the inventory component on render, pls fix:
Screenshot 2023-10-20 at 10 16 33

Pls also provide QA with screenshots of all of the rendered components mentioned in the [original issue](https://github.com/ubiquity/ubiquity-dollar/issues/807).

**Comment by bojan07** [comment]
image

**Comment by bojan07** [comment]
image
This error is just normal error that is generated because there is no pair in curve meta pool, as I know

**Comment by bojan07** [comment]
image
DAI-USDT returns the value, but DOLLAR-USDC returns error.

**Comment by bojan07** [comment]
> There is an error in the inventory component on render, pls fix: Screenshot 2023-10-20 at 10 16 33
> 
When testing after running anvil testnet and deploying contracts, this kind of error will not happen. I can't find this kind of error in my side

**Comment by bojan07** [comment]
Just removed the possibility of that error case in [this commit](https://github.com/ubiquity/ubiquity-dollar/pull/824/commits/070fa7206312994fb517807b6b1064b83a799ec2)

**Comment by bojan07** [comment]
> image DAI-USDT returns the value, but DOLLAR-USDC returns error.

If new pairs were made with newly deployed Dollar token in the curve pool, this kind of errors will be disappeared

**Comment by bojan07** [comment]
image

**Comment by molecula451** [comment]
> This error is just normal error that is generated because there is no pair in curve meta pool, as I know

there is a pair

**Comment by molecula451** [comment]
there is a pair, you can mock a Pair to test the components too

**Comment by bojan07** [comment]
Can I know the already deployed uAD address?

**Comment by bojan07** [comment]
> Can I know the already deployed uAD address?

Found that address
https://etherscan.io/address/0x0f644658510c95cb46955e55d7ba9dda9e9fbec6
Passed `get_coin_indices` function, but now new error - `INVALID TOKEN` related to twap oracle
image

**Comment by molecula451** [comment]
> > Can I know the already deployed uAD address?
> 
> Found that address https://etherscan.io/address/0x0f644658510c95cb46955e55d7ba9dda9e9fbec6 Passed `get_coin_indices` function, but now new error - `INVALID TOKEN` related to twap oracle image

test with diamond uAD

**Comment by bojan07** [comment]
> `INVALID TOKEN` related to twap oracle

Had a look at `TWAPOracleDollar3poolFacet` contract. Because I have tested  with old dollar address in stead of newly deployed dollar address, it doesn't exist on `twap oracle storage`.. so invalid token..
How can I mock the pair?

**Comment by molecula451** [comment]
> > `INVALID TOKEN` related to twap oracle
> 
> Had a look at `TWAPOracleDollar3poolFacet` contract. Because I have tested with old dollar address in stead of newly deployed dollar address, it doesn't exist on `twap oracle storage`.. so invalid token.. How can I mock the pair?

the tests have this interactions, theres mocking pairs etc here https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/test try to achieve something similar to test the components

**Comment by molecula451** [comment]
> > > `INVALID TOKEN` related to twap oracle
> > 
> > 
> > Had a look at `TWAPOracleDollar3poolFacet` contract. Because I have tested with old dollar address in stead of newly deployed dollar address, it doesn't exist on `twap oracle storage`.. so invalid token.. How can I mock the pair?
> 
> the tests have these interactions, theres mocking pairs etc here https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/test try to achieve something similar to test the components

**Comment by bojan07** [comment]
image
Just tested by setting the return value of consult function of `twap oracle` manually, working well

**Comment by molecula451** [comment]
> image Just tested by setting the return value of consult function of `twap oracle` manually, working well

nice please show us some console.log beautiness

**Comment by molecula451** [comment]
>  Just tested by setting the return value of consult function of `twap oracle` manually, working well

but i still think this is not what we really want, try to achieve the result without manually setting

**Comment by bojan07** [comment]
> nice please show us some console.log beautiness

image

**Comment by bojan07** [comment]
> > Just tested by setting the return value of consult function of `twap oracle` manually, working well
> 
> but i still think this is not what we really want, try to achieve the result without manually setting

Then, We should update deployment script(when running yarn deploy:development) to create pairs in curve meta pool.. not quite needed in this QA testing stage

**Comment by molecula451** [comment]
please test the components mock a pair you dont need to change deployment script in this PR what you need to do is test the components which whatever way possible

**Comment by bojan07** [comment]
> please test the components mock a pair you dont need to change deployment script in this PR what you need to do is test the components which whatever way possible

Yeah, but mocking pair involves solidity works, not ui works? I think we don't have to do such process to test small ui stuff

**Comment by bojan07** [comment]
@molecula451 I'm quite new to ubiquity-dollar project and only solved ui related 3 issues. So, involving solidity works might require much time for me.. I think mocking Pair is not a MUST for QA of the PR.. and QA was provided by hardcoding, so hope to skip mocking pair

**Comment by rndquu** [comment]
@bojan07 pls provide QA screenshots for the following components:
- https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/layout/inventory.tsx
- https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/debt-coupon-deposit.tsx
- https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/ucr-nft-redeem.tsx
- https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/ucr-redeem.tsx

**Comment by bojan07** [comment]
> @bojan07 pls provide QA screenshots for the following components:
> 
> * https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/layout/inventory.tsx
> * https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/debt-coupon-deposit.tsx
> * https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/ucr-nft-redeem.tsx
> * https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/ucr-redeem.tsx

inventory component is imported in each page, so homepage screenshot shows it

image

And other 3 components are imported in credits page(As we checked in the last pr, only `Network error` come out due to Uniswap router package)

image

**Comment by bojan07** [comment]
https://github.com/ubiquity/ubiquity-dollar/blob/97989f176ebcbf0e726d3d5aa5029e85fc1f15d0/packages/dapp/components/redeem/debt-coupon.tsx
This component also has been updated regarding to name convention. and it is shown in `credit-nft` page
image

**Comment by rndquu** [comment]
@bojan07 

> inventory component is imported in each page, so homepage screenshot shows it

Yes, it is imported in each page but there are no any screenshots where it is rendered with balances, pls provide one.

**Comment by bojan07** [comment]
> @bojan07
> 
> > inventory component is imported in each page, so homepage screenshot shows it
> 
> Yes, it is imported in each page but there are no any screenshots where it is rendered with balances, pls provide one.

image

**Comment by bojan07** [comment]
image

**Comment by bojan07** [comment]
Fixed broken ui
image

File: packages/dapp/components/utils/contracts.ts [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { Provider } from "@ethersproject/providers";
...
 // ARTIFACTS
-import _DebtCoupon from "@ubiquity/contracts/out/CreditNft.sol/CreditNft.json";
-import _DebtCouponManager from "@ubiquity/contracts/out/CreditNftManagerFacet.sol/CreditNftManagerFacet.json";
+import _CreditNft from "@ubiquity/contracts/out/CreditNft.sol/CreditNft.json";
+import _CreditNftManager from "@ubiquity/contracts/out/CreditNftManagerFacet.sol/CreditNftManagerFacet.json";
 import _DollarMintCalculator from "@ubiquity/contracts/out/DollarMintCalculatorFacet.sol/DollarMintCalculatorFacet.json";
 import _ERC1155Ubiquity from "@ubiquity/contracts/out/ERC1155Ubiquity.sol/ERC1155Ubiquity.json";
-import _ICouponsForDollarsCalculator from "@ubiquity/contracts/out/ICreditNftRedemptionCalculator.sol/ICreditNftRedemptionCalculator.json";
-import _IUARForDollarsCalculator from "@ubiquity/contracts/out/ICreditRedemptionCalculator.sol/ICreditRedemptionCalculator.json";
+import _INftsForDollarsCalculator from "@ubiquity/contracts/out/ICreditNftRedemptionCalculator.sol/ICreditNftRedemptionCalculator.json";
```
Comment (rndquu) [inline packages/dapp/components/utils/contracts.ts]:
```suggestion
import _ICreditNftRedemptionCalculator from "@ubiquity/contracts/out/ICreditNftRedemptionCalculator.sol/ICreditNftRedemptionCalculator.json";
```

File: packages/dapp/components/utils/contracts.ts [hunk 2]
```diff
@@ -3,12 +3,12 @@ import { Provider } from "@ethersproject/providers";
...
+import _ICreditForDollarsCalculator from "@ubiquity/contracts/out/ICreditRedemptionCalculator.sol/ICreditRedemptionCalculator.json";
```
Comment (rndquu) [inline packages/dapp/components/utils/contracts.ts]:
```suggestion
import _ICreditRedemptionCalculator from "@ubiquity/contracts/out/ICreditRedemptionCalculator.sol/ICreditRedemptionCalculator.json";
```

File: packages/dapp/components/utils/contracts.ts [hunk 3]
```diff
@@ -139,12 +139,12 @@ export const getDollarMintCalculatorContract = (address: string, provider: Provi
   return getContract(_DollarMintCalculator.abi, address, provider) as DollarMintCalculatorFacet;
 };
 
-export const getICouponsForDollarsCalculatorContract = (address: string, provider: Provider) => {
-  return getContract(_ICouponsForDollarsCalculator.abi, address, provider) as CreditNftRedemptionCalculatorFacet;
+export const getINftsForDollarsCalculatorContract = (address: string, provider: Provider) => {
```
Comment (rndquu) [inline packages/dapp/components/utils/contracts.ts]:
```suggestion
export const getCreditNftRedemptionCalculatorContract = (address: string, provider: Provider) => {
```

File: packages/dapp/components/utils/contracts.ts [hunk 4]
```diff
@@ -139,12 +139,12 @@ export const getDollarMintCalculatorContract = (address: string, provider: Provi
...
+  return getContract(_INftsForDollarsCalculator.abi, address, provider) as CreditNftRedemptionCalculatorFacet;
 };
 
-export const getIUARForDollarsCalculatorContract = (address: string, provider: Provider) => {
-  return getContract(_IUARForDollarsCalculator.abi, address, provider) as CreditRedemptionCalculatorFacet;
+export const getICreditForDollarsCalculatorContract = (address: string, provider: Provider) => {
```
Comment (rndquu) [inline packages/dapp/components/utils/contracts.ts]:
```suggestion
export const getCreditRedemptionCalculatorContract = (address: string, provider: Provider) => {
```

**Review by rndquu — CHANGES_REQUESTED** [review]
@bojan07 Overall look fine, pls fix some naming issues

File: packages/dapp/components/utils/contracts.ts (hunk 4 above)
Comment (bojan07) [reply packages/dapp/components/utils/contracts.ts]: done it

**Review by rndquu — APPROVED** [review]

**Comment by bojan07** [comment]
@molecula451 how is the review going? I'm waiting for your review

**Review by molecula451 — APPROVED** [review]
This PR only adds cosmetical diamond changes but importants and needed, there doesn't seem to be major refactoring only the changes and deletes of managed-manager hook (which was planned) necessary for the well-being of the app and furthercoming diamond connections

**Pull request merged**